lib/dotcom_web/templates/schedule/_alerts.html.heex

```diff
@@ -0,0 +1,32 @@
+<div class="page-section m-schedule-line">
+  <% subway? =
+    Map.get(@conn.assigns, :breadcrumbs, [])
+    |> Enum.any?(fn %{text: text} -> text === "Subway" end) %>
```

Comment:
Could we call `group_alerts` out here and pass that into both `titles_by_effect` and `alerts_by_effect`?

Right now, we call `group_alerts` three times - twice in `alerts_by_effect` and once in `titles_by_effect`. It would be helpful to reduce that to just once.

This would involve having `titles_by_effect` call `Enum.count` on `@grouped_alerts` to get the counts, rather than just looking them up, but then we could get rid of `group_counts/1`, which feels like a win.

Reply:
The calls to group and count probably take microseconds. Having them split out makes them more flexible for future use.

Reply:
See above, it takes 99 *millionths* of a second to group alerts...which is why I chose flexibility and communicability over trying to minimize computations.

Reply:
I was never concerned about the computation - I was more concerned about the cognitive overhead of each component redoing the same computation, in some cases multiple times. 

Non-blocking.

Reply:
Right, but then let's say you *did* want the count somewhere else. It just so happens that counting needs grouping. By giving the consumer a way to get `group_counts()` without having to call `group_alerts()` first, we give them a better experience. It would be weird to have a public function that is called `count_already_grouped_alerts(alerts)`.